paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -1054,17 +1054,77 @@ class PaymentSheet internal constructor(
          * The three-letter ISO 4217 alphabetic currency code, e.g. "USD" or "EUR".
          * Required in order to support Google Pay when processing a Setup Intent.
          */
-        val currencyCode: String? = null
+        val currencyCode: String? = null,
+        /**
+         * The Google Pay button type to use. Set to "Pay" by default.
+         *
+         * See [Google's documentation](https://developers.google.com/pay/api/android/reference/request-objects#ButtonOptions) for more information on button types.
+         */
+        val buttonType: ButtonType = ButtonType.Pay
     ) : Parcelable {
         constructor(
             environment: Environment,
             countryCode: String
-        ) : this(environment, countryCode, null)
+        ) : this(environment, countryCode, null, ButtonType.Pay)
+
+        constructor(
+            environment: Environment,
+            countryCode: String,
+            currencyCode: String?
+        ) : this(environment, countryCode, currencyCode, ButtonType.Pay)
 
         enum class Environment {
             Production,
             Test
         }
+
+        @Suppress("MaxLineLength")
+        /**
+         * Google Pay button type options
+         *
+         * See [Google's documentation](https://developers.google.com/pay/api/android/reference/request-objects#ButtonOptions) for more information on button types.
```

Comment:
I've had to suppress the warning here because of comment length despite other comments having similar length. Wondering if we should exclude comments for max line length check.

Reply:
I'm guessing others are ignored by running `./gradlew detektbaseline`, and those live in https://github.com/stripe/stripe-android/blob/33ae56f2791732e56d43c529af9de430086bfcfe/paymentsheet/detekt-baseline.xml#L1

Suppressing either way is fine by me.